stubs/invoke/invoke/context.pyi

```diff
@@ -1,15 +1,16 @@
 from contextlib import AbstractContextManager
 
 from .config import Config, DataProxy
+from .runners import Result
 
 class Context(DataProxy):
     def __init__(self, config: Config | None = ...) -> None: ...
     @property
     def config(self) -> Config: ...
     @config.setter
     def config(self, value: Config) -> None: ...
-    def run(self, command: str, **kwargs): ...
-    def sudo(self, command: str, *, password: str = ..., user: str = ..., **kwargs): ...
+    def run(self, command: str, **kwargs) -> Result: ...
+    def sudo(self, command: str, **kwargs) -> Result: ...
```

Comment:
Both `run` and `sudo` call `Runner.run`:
- https://github.com/pyinvoke/invoke/blob/ce85bdf11ae9a88e18e5d669310ff0bfd4338725/invoke/context.py#L219
- https://github.com/pyinvoke/invoke/blob/ce85bdf11ae9a88e18e5d669310ff0bfd4338725/invoke/context.py#L102

`Runner.run` returns different types in different cases: https://github.com/python/typeshed/blob/6950c3237065e6e2a9b64810765fec716252d52a/stubs/invoke/invoke/runners.pyi#L18-L120

We decide what to return based on `disown: bool` and `asynchronous` args. How can we be sure that it is always `Result` here?

Reply:
In __init__, I saw that run and sudo returns Result. Promise is also a Result. What makes this different?

Reply:
It also returns `None` and `Any`

Reply:
Yes but here it's only returning result https://github.com/python/typeshed/blob/6950c3237065e6e2a9b64810765fec716252d52a/stubs/invoke/invoke/__init__.pyi#L36-L37